static/js/utils/subject_page_utils.ts

```diff
@@ -210,3 +210,74 @@ export function trimCategory(
   }
   return pageConfig;
 }
+
+/**
+ * Add "Per Capita" to a chart title.
+ *
+ * If the "date" replacement string is present, will add "Per Capita in" inside
+ * the same parentheses as the date. If date is not present, will add
+ * "(Per Capita)" to the end of the title.
+ *
+ * Assumes chart titles have the format <StatVarName> (${date}). Used for
+ * updating the chart titles when the "See per capita" checkbox is checked.
+ *
+ * @param title title of the chart to edit
+ * @param dateString format of the date replacement string to look for
+ * @returns new chart title that includes the string "Per Capita"
+ */
+export function addPerCapitaToTitle(
+  title: string,
+  dateString = "date"
+): string {
+  if (title.includes(`(\${${dateString}})`)) {
+    const regex = new RegExp(`\\(\\$\\{${dateString}\\}\\)`);
+    return title.replace(regex, `(Per Capita in $\{${dateString}})`);
+  }
+  return title + " (Per Capita)";
+}
+
+/**
+ * Add "Per Capita" to a chart title that compares two stat vars
+ *
+ * Adds "Per Capita" to the variables that have a denom in their spec. If xDate
+ * or yDate replacement strings are present, will add "Per Capita" inside the
+ * same parentheses as the corresponding variable date. If xDate or yDate is
+ * not present, will add "(Per Capita)" after the variable name instead.
+ *
+ * Assumes the first stat var in statVarSpecs is shown on the y-axis, and
+ * the second stat var in statVarSpecs is shown on the x-axis. Assumes the
+ * format of the title originally is "<y-axis stat var> Vs. <x-axis stat var> in
+ * <placeType> of <parentPlace>" or "<y-axis stat var> Vs. <x-axis stat var>".
+ *
+ * @param title title of chart to edit
+ * @param statVarSpecs stat vars being plotted by the chart
+ * @returns new chart title that includes "Per Capita" for vars with denom
+ */
+export function addPerCapitaToVersusTitle(
+  title: string,
+  statVarSpecs: StatVarSpec[]
+): string {
+  // Split title into constituent parts via regex groups:
+  //   xVar -> x-axis stat var
+  //   yVar -> y-axis stat var
+  //   vs -> " Vs. " or " vs. " or " Vs " or " vs "
+  //   location -> "in <placeType> of <place>", if present
+  const regex =
+    /^(?<yVar>.*)(?<vs>\s[Vv]s\.?\s)(?<xVar>.*)(?<location>\s+in\s.*)?$/;
```

Comment:
Parsing on "vs." sounds potentially brittle-this could break with localized scatterplot titles.

Instead why not do something like:

```
let mutableTitle = title;
if (statVarSpecs?.[0].denom) {
  mutableTitle = addPerCapitaToTitle(mutableTitle, "yDate");
}
if (statVarSpecs?.[1].denom) {
  mutableTitle = addPerCapitaToTitle(mutableTitle, "xDate");
}
return mutableTitle;
```

Reply:
The approach you suggested breaks when only one variable is per capita and xDate/yDate are not present. "(Per Capita)" without the date would be added at the end of the title, after "in <placeType> of <place>", and it'd be unclear which variable is the one that is per capita.

I agree that the current approach is brittle. I think ideally we'd go one step further and make tiles generate their own titles based on the vars/places passed in and skip this post-hoc regex parsing altogether. What do you think?

Reply:
Makes sense. and agree the tiles should be generating their own titles by default.